bodo/pandas/frame.py

```diff
@@ -1033,7 +1039,12 @@ def _update_setitem_internal_state(
             # Update internal data manager (e.g. insert a new block or update an
             # existing one). See:
             # https://github.com/pandas-dev/pandas/blob/0691c5cf90477d3503834d983f69350f250a6ff7/pandas/core/frame.py#L4481
-            super().__setitem__(key, head_val)
+            if new_column:
+                self._mgr.insert(
+                    len(self._info_axis) - 1, key, *self._sanitize_column(head_val)
+                )
+            else:
+                super().__setitem__(key, head_val)
```

Comment:
What was the problem with `super().__setitem__` in this case? Do you have a reproducer? Using Pandas is desirable if possible to make sure all the internal details are set and we are more future proof.

Reply:
__setitem__ has a check that requires length which causes a premature execution of the plan.

[Function that does len check.](https://github.com/pandas-dev/pandas/blob/0f437949513225922d851e9581723d82120684a6/pandas/core/frame.py#L4108)

Reply:
Else may have the problem.  It is untested.  I'll try to add a test.